### PR TITLE
Feature/add fees

### PR DESCRIPTION
**Pull request by talekhinezh**

### CODE REVIEW

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/api/handler/DeveloperHandler.java [hunk 1]
```diff
@@ -128,14 +131,28 @@ public JSONObject handleParseTxn(JSONObject request) {
...
+							break;
+						case "node":
+							pair = NodeAddressing.parseUnknownHrp(address);
+							break;
+						case "validator":
+							pair = ValidatorAddressing.parseUnknownHrp(address);
+							break;
+						default:
+							throw new IllegalArgumentException("type must be: [account|node|validator]");
+					}
```
Comment (siy) [inline radixdlt-core/radixdlt/src/main/java/com/radixdlt/api/handler/DeveloperHandler.java:153]: I think code will be much more readable if `switch` will be extracted into dedicated method.

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/api/handler/DeveloperHandler.java [hunk 2]
```diff
@@ -128,14 +131,28 @@ public JSONObject handleParseTxn(JSONObject request) {
...
+					var hrp = pair.getFirst();
+					var pubKey = pair.getSecond();
 					return jsonObject()
 						.put("hrp", hrp)
 						.put("public_key", Bytes.toHexString(pubKey.getCompressedBytes()));
```
Comment (siy) [inline radixdlt-core/radixdlt/src/main/java/com/radixdlt/api/handler/DeveloperHandler.java:158]: Alternative version:
```
return pair.map((hrp, pubKey) -> 
    jsonObject()
        .put("hrp", hrp)
	.put("public_key", Bytes.toHexString(pubKey.getCompressedBytes())));
```

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/api/handler/DeveloperHandler.java [hunk 3]
```diff
@@ -147,14 +164,29 @@ public JSONObject handleParseAddress(JSONObject request) {
...
+							break;
+						case "node":
+							address = NodeAddressing.bech32(hrp).of(publicKey);
+							break;
+						case "validator":
+							address = ValidatorAddressing.bech32(hrp).of(publicKey);
+							break;
+						default:
+							throw new IllegalArgumentException("type must be: [account|node|validator]");
+					}
```
Comment (siy) [inline radixdlt-core/radixdlt/src/main/java/com/radixdlt/api/handler/DeveloperHandler.java:188]: I think it makes sense to extract this logic into dedicated method.

**Review by siy — APPROVED** [review]
LGTM with comments

**Review by iamyulong — APPROVED** [review]
Looks good to me, but I have one question regarding encoding of optional fields:
- Why do we need to encode a zero-value for nones? Is it to make substates of fixed length for better indexing?
- I was also thinking maybe we should index based on fields (rather than prefix) for flexibility.

**Pull request merged**